### ISSUE TITLE
bashobfus fails to minify bash function names

### ISSUE THREAD

**Issue by kormoc** [Aralhach/bashobfus#1]
This leaves readable code in any complex bash script!

**Comment by Aralhach** [comment]
Yes, that would be a missing feature... I'm not sure how I would detect all the function appearances without a better lexical parsing of the language.  I would like to upgrade this to parse out the syntactical elements of Bash and do the substitutions in a smarter way, minify the script into a single line, and sweet stuff like that.

It's currently more of a hacky regex substituter that worked for the script I ran it through. Thank you very much for testing the script and for your input! :) I appreciate it!
